bindings.c

```diff
@@ -5874,6 +5878,275 @@ int proc_read(const char *path, char *buf, size_t size, off_t offset,
 	}
 }
 
+static int sys_devices_system_cpu_read(char *buf, size_t size, off_t offset,
+		struct fuse_file_info *fi)
+{
+	struct fuse_context *fc = fuse_get_context();
+	struct file_info *d = (struct file_info *)fi->fh;
+	char *cache = d->buf;
+	ssize_t total_len = 0;
+	if (offset){
+		if (!d->cached)
+			return 0;
+		if (offset > d->size)
+			return -EINVAL;
+		int left = d->size - offset;
+		total_len = left > size ? size: left;
+		memcpy(buf, cache + offset, total_len);
+		return total_len;
+	}
+
+	total_len = snprintf(d->buf, d->buflen, "0-1\n");
+	if (total_len < 0 || total_len >=  d->buflen){
+		lxcfs_error("%s\n", "failed to write to cache");
+		return 0;
+	}
+
+	d->size = (int)total_len;
+	d->cached = 1;
+
+	if (total_len > size) total_len = size;
+
+	memcpy(buf, d->buf, total_len);
+	return total_len;
+}
+
+static int sys_devices_system_cpu_online_read(char *buf, size_t size, off_t offset,
+		struct fuse_file_info *fi)
+{
+	struct fuse_context *fc = fuse_get_context();
+	struct file_info *d = (struct file_info *)fi->fh;
+	char *cache = d->buf;
+	char *cg;
+	char *cpuset = NULL;
+	bool use_view;
+
+        int max_cpus = 0;
+	ssize_t total_len = 0;
+	if (offset){
+		if (!d->cached)
+			return 0;
+		if (offset > d->size)
+			return -EINVAL;
+		int left = d->size - offset;
+		total_len = left > size ? size: left;
+		memcpy(buf, cache + offset, total_len);
+		return total_len;
+	}
+	pid_t initpid = lookup_initpid_in_store(fc->pid);
+	if (initpid <= 0)
+		initpid = fc->pid;
+	cg = get_pid_cgroup(initpid, "cpuset");
+	if (!cg)
+		return read_file("/sys/devices/system/cpu/online", buf, size, d);
+	prune_init_slice(cg);
+
+	cpuset = get_cpuset(cg);
+	if (!cpuset)
+		goto err;
+
+	use_view = use_cpuview(cg);
+
+	if (use_view)
+		max_cpus = max_cpu_count(cg);
+
+
+	if (max_cpus == 0)
+		return read_file("/sys/devices/system/cpu/online", buf, size, d);
+	if (max_cpus > 1)
+		total_len = snprintf(d->buf, d->buflen, "0-%d\n", max_cpus - 1);
+	else
+		total_len = snprintf(d->buf, d->buflen, "0\n");
+	if (total_len < 0 || total_len >=  d->buflen){
+		lxcfs_error("%s\n", "failed to write to cache");
+		return 0;
+	}
+
+	d->size = (int)total_len;
+	d->cached = 1;
+
+	if (total_len > size) total_len = size;
+
+	memcpy(buf, d->buf, total_len);
+err:
+	free(cpuset);
+	free(cg);
+	return total_len;
+}
+
+static off_t get_sysfile_size(const char *which)
+{
+	FILE *f = fopen(which, "r");
+	char *line = NULL;
+	size_t len = 0;
+	ssize_t sz, answer = 0;
+	if (!f)
+		return 0;
```

Comment:
please do not call a function right when you declare the variable if you check the error condition later, instead do
FILE *f;
.
.
.

f = fopen(which, "r");
if (!f)